tools/viz.py

```diff
@@ -14,16 +15,46 @@ def __init__(
         ice_thickness: str,
         x: str = "x",
         y: str = "y",
-        time: str = "time",
-        topo_bedrock: str = "topo_bed_rock",
+        x_border: int = 100,
+        y_border: int = 100,
+        zoom: float = 1.,
+        azimuth: float = 0.,
+        elevation: float = 10.,
+        roll: float = 0.,
     ):
-        self.dataset = dataset
         self.x = x
         self.y = y
-        self.time = time
-
-        self.da_topo = dataset[topo_bedrock]
-        self.da_glacier_surf = self.da_topo + dataset[ice_thickness]
+        self.zoom = zoom
+        self.azimuth = azimuth
+        self.elevation = elevation
+        self.roll = roll
+
+        # resize map to given border values
+        x_middle_point = int(len(dataset[self.x]) / 2)
+        y_middle_point = int(len(dataset[self.y]) / 2)
+        self.dataset = dataset.isel({self.x: slice(x_middle_point - x_border,
+                                                   x_middle_point + x_border),
+                                     self.y: slice(y_middle_point - y_border,
+                                                   y_middle_point + y_border)
+                                     }).load()
+
+        # check if we should use complete years or monthly timesteps
```

Comment:
This has changed in the new API (sorry). normally the OGGM files should have the necessary data available (including bedrock)